recipes/libftdi/1.x/conanfile.py

```diff
@@ -1,103 +1,107 @@
+        if is_msvc(self) and self.options.use_streaming:
             raise ConanInvalidConfiguration("VS doesn't not compile with enabled option use_streaming")
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
-        cmake = self._configure_cmake()
+        copy(self, "COPYING.LIB", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        copy(self, "LICENSE", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        if self.options.build_eeprom_tool or self.options.enable_cpp_wrapper:
+            copy(self, "COPYING.GPL", self.source_folder, os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
-        lib_folder = os.path.join(self.package_folder, "lib",)
-        tools.rmdir(os.path.join(lib_folder, "cmake"))
-        tools.rmdir(os.path.join(lib_folder, "pkgconfig"))
+        lib_folder = os.path.join(self.package_folder, "lib")
+        rmdir(self, os.path.join(lib_folder, "cmake"))
+        rmdir(self, os.path.join(lib_folder, "pkgconfig"))
 
     def package_info(self):
-        self.cpp_info.names["cmake_find_package"] = "LibFTDI1"
-        self.cpp_info.names["cmake_find_package_multi"] = "LibFTDI1"
-        self.cpp_info.names["pkgconfig"] = "libftdi1"
+        # Remove "self.cpp_info.filenames.." statements in Conan V2
+        self.cpp_info.filenames['cmake_find_package'] = "LibFTDI1"
+        self.cpp_info.filenames['cmake_find_package_multi'] = "LibFTDI1"
 
-        self.cpp_info.components["ftdi"].names["pkg_config"] = "libftdi1"
+        self.cpp_info.set_property("cmake_file_name", "LibFTDI1")
+        self.cpp_info.components["ftdi"].set_property("pkg_config_name", "libftdi1")
         self.cpp_info.components["ftdi"].libs = ["ftdi1"]
         self.cpp_info.components["ftdi"].requires = ["libusb::libusb"]
         self.cpp_info.components["ftdi"].includedirs.append(os.path.join("include", "libftdi1"))
 
-        self.cpp_info.components["ftdipp"].names["pkg_config"] = "libftdi1pp"
-        self.cpp_info.components["ftdipp"].libs = ["ftdipp1"]
-        self.cpp_info.components["ftdipp"].requires = ["ftdi", "boost::headers"]
-        self.cpp_info.components["ftdipp"].includedirs.append(os.path.join("include", "libftdipp1"))
+        if self.options.enable_cpp_wrapper:
+            self.cpp_info.components["ftdipp"].set_property("pkg_config_name", "libftdipp1")
+            self.cpp_info.components["ftdipp"].libs = ["ftdipp1"]
+            self.cpp_info.components["ftdipp"].requires = ["ftdi", "boost::headers"]
```

Comment:
This will fix those lint warnings.

```suggestion
            self.cpp_info.components["ftdipp"].requires = ["ftdi", "boost::headers"]
            if self.settings.os in ["Linux", "FreeBSD"]:
                self.cpp_info.components["ftdipp"].system_libs.append("m")
```